packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -954,11 +958,11 @@ class _PopupMenuRoute<T> extends PopupRoute<T> {
       removeBottom: true,
       removeLeft: true,
       removeRight: true,
-      child: Builder(
-        builder: (BuildContext context) {
+      child: LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) {
           return CustomSingleChildLayout(
             delegate: _PopupMenuRouteLayout(
-              position,
+              positionBuilder?.call() ?? position!,
```

Comment:
I think we need to pass the `context` to the `positionBuilder` here. Probably the `constraints` as well, although I'm not exactly sure whose contraints they are and whether they're meaningful. Otherwise the callback can not effectively do much.

Reply:
The context you're suggesting to add would be the route's context, not the PopupMenuButton's context. The route's context isn't useful for positioning since we already have access to the button's context through PopupMenuButtonState. The current implementation with `RelativeRect Function()` is sufficient.

Reply:
But what about the caller of `showMenu`? In the simplest case we'd like any `context` to get the screen parameters, but even this is not available if a stateless widget is to call `showMenu`.